inception-imls-opennlp/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/opennlp/doccat/OpenNlpDoccatRecommenderTest.java

```diff
@@ -0,0 +1,233 @@
+
+import static java.util.Arrays.asList;
+import static org.apache.uima.fit.factory.AnalysisEngineFactory.createEngine;
+import static org.apache.uima.fit.factory.CollectionReaderFactory.createReader;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.LineIterator;
+import org.apache.uima.UIMAException;
+import org.apache.uima.analysis_engine.AnalysisEngine;
+import org.apache.uima.cas.CAS;
+import org.apache.uima.collection.CollectionException;
+import org.apache.uima.collection.CollectionReader;
+import org.apache.uima.fit.factory.JCasFactory;
+import org.apache.uima.fit.util.JCasUtil;
+import org.apache.uima.jcas.JCas;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.tudarmstadt.ukp.clarin.webanno.model.AnnotationLayer;
+import de.tudarmstadt.ukp.dkpro.core.api.datasets.Dataset;
+import de.tudarmstadt.ukp.dkpro.core.api.datasets.DatasetFactory;
+import de.tudarmstadt.ukp.dkpro.core.api.io.JCasResourceCollectionReader_ImplBase;
+import de.tudarmstadt.ukp.dkpro.core.api.ner.type.NamedEntity;
+import de.tudarmstadt.ukp.dkpro.core.api.resources.CompressionUtils;
+import de.tudarmstadt.ukp.dkpro.core.api.segmentation.type.Sentence;
+import de.tudarmstadt.ukp.dkpro.core.testing.DkproTestContext;
+import de.tudarmstadt.ukp.dkpro.core.tokit.BreakIteratorSegmenter;
+import de.tudarmstadt.ukp.inception.recommendation.api.evaluation.DataSplitter;
+import de.tudarmstadt.ukp.inception.recommendation.api.evaluation.IncrementalSplitter;
+import de.tudarmstadt.ukp.inception.recommendation.api.evaluation.PercentageBasedSplitter;
+import de.tudarmstadt.ukp.inception.recommendation.api.model.Recommender;
+import de.tudarmstadt.ukp.inception.recommendation.api.recommender.RecommenderContext;
+import de.tudarmstadt.ukp.inception.recommendation.api.type.PredictedSpan;
+
+public class OpenNlpDoccatRecommenderTest
+{
+    private static File cache = DkproTestContext.getCacheFolder();
+    private static DatasetFactory loader = new DatasetFactory(cache);
+
+    private RecommenderContext context;
+    private Recommender recommender;
+    private OpenNlpDoccatRecommenderTraits traits;
+
+    @Before
+    public void setUp()
+    {
+        context = new RecommenderContext();
+        recommender = buildRecommender();
+        traits = new OpenNlpDoccatRecommenderTraits();
+    }
+
+    @Test
+    public void thatTrainingWorks() throws Exception
+    {
+        OpenNlpDoccatRecommender sut = new OpenNlpDoccatRecommender(recommender, traits);
+        List<CAS> casList = loadDevelopmentData();
+
+        sut.train(context, casList);
+
+        assertThat(context.get(OpenNlpDoccatRecommender.KEY_MODEL))
+            .as("Model has been set")
+            .isNotNull();
```

Comment:
I think that this test does not make sense, as you return now an Optional

Reply:
Was copy-pasted from the other test - probably needs to be changed everywhere then.